pkg/agent/plugin/svidstore/aws/aws.go

```diff
@@ -0,0 +1,289 @@
+	}
+
+	// Call DescribeSecret to retrieve the details of the secret
+	// and be able to determine if the secret exists
+	secretDesc, err := p.smClient.DescribeSecret(ctx, &secretsmanager.DescribeSecretInput{
+		SecretId: aws.String(secretID),
+	})
+	if err != nil {
+		var resourceNorFoundErr *types.ResourceNotFoundException
+		if errors.As(err, &resourceNorFoundErr) {
+			// Secret not found, creating one with provided `name`
+			resp, err := createSecret(ctx, p.smClient, secretBinary, opt)
+			if err != nil {
+				return nil, err
+			}
+			p.log.With("version_id", aws.StringValue(resp.VersionId)).With("arn", aws.StringValue(resp.ARN)).With("name", aws.StringValue(resp.Name)).Debug("Secret created")
+
+			return &svidstorev1.PutX509SVIDResponse{}, nil
+		}
+
+		// It must not happens.
+		return nil, status.Errorf(codes.Internal, "failed to describe secret: %v", err)
+	}
+
+	// Validate if secret is deleted, and restore it
+	if secretDesc.DeletedDate != nil {
+		resp, err := p.smClient.RestoreSecret(ctx, &secretsmanager.RestoreSecretInput{
+			SecretId: aws.String(secretID),
+		})
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "failed to restore secret %q: %v", secretID, err)
+		}
+		p.log.With("arn", aws.StringValue(resp.ARN)).With("name", aws.StringValue(resp.Name)).Debug("Secret restored")
+	}
+
+	// Validate that the secret has the 'spire-svid' tag. This tag is used to distinguish the secrets
+	// that have SVID information handled by SPIRE
+	if ok := validateTag(secretDesc.Tags); !ok {
+		return nil, status.Error(codes.InvalidArgument, "secret does not contain the 'spire-svid' tag")
+	}
+
+	putResp, err := p.smClient.PutSecretValue(ctx, &secretsmanager.PutSecretValueInput{
+		SecretId:     secretDesc.ARN,
+		SecretBinary: secretBinary,
+	})
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to put secret value: %v", err)
+	}
+
+	p.log.With("version_id", aws.StringValue(putResp.VersionId)).With("arn", aws.StringValue(putResp.ARN)).With("name", aws.StringValue(putResp.Name)).Debug("Secret value updated")
+	return &svidstorev1.PutX509SVIDResponse{}, nil
+}
+
+func (p *SecretsManagerPlugin) DeleteX509SVID(ctx context.Context, req *svidstorev1.DeleteX509SVIDRequest) (*svidstorev1.DeleteX509SVIDResponse, error) {
+	opt, err := optionsFromSecretData(req.Metadata)
+	if err != nil {
+		return nil, err
+	}
+
+	// Use arn if it is configured if not use name
+	secretID := opt.name
+	if opt.arn != "" {
+		secretID = opt.arn
+	}
+
+	// Call DescribeSecret to retrieve the details of the secret
+	// and be able to determine if the secret exists
+	secretDesc, err := p.smClient.DescribeSecret(ctx, &secretsmanager.DescribeSecretInput{
+		SecretId: aws.String(secretID),
+	})
+	if err != nil {
+		var resourceNorFoundErr *types.ResourceNotFoundException
+		if errors.As(err, &resourceNorFoundErr) {
+			p.log.With("secret_id", secretID).Debug("Secret already deleted")
```

Comment:
Do we expect this to be a normal situation? If not, I think this should be a warning and worded differently, like "Secret does not exist"